File: c programing/tests/q1.c
// 1. Write a program to count number of words in a string.
#include<stdio.h>
int countWords(char[]);
void main(){
     char str[]="hello word i am student of firstbit";
     int count=countWords(str);
     printf("words present in string = %d\n",count);
}
int countWords(char str[]){
   int len=0;
   while(str[len]!='\0'){
      len++;
   }

   if (len == 0)
   {
      printf("String is empty");
      return 0;
   }

   int start=0;
   int end=len-1;
   while(str[start]==' ')start++;
   while(str[end]==' ')end--;

   int count=0;
   
   for(int i=start;i<end;i++){
      if(str[i]==' '){
         count++;
      }
   }
   count++;
   return count;
}

File: java/practice/q1.java
// Write a program to count the number of words in a given sentence

import java.util.Scanner;

public class q1{
   public static void main(String[] args) {
      String s;
      Scanner sc=new Scanner(System.in);
      StringOperation so=new StringOperation();
      System.out.println("enter String");
      s=sc.nextLine();
      System.out.println(so.countWords(s));
   }
}
class StringOperation{
   public int countWords(String st){
      if(st.length()==0){
         return 0;
      }
      int len=1;
      String temp=st.trim();
      for(int i=0;i<temp.length();i++){
         if(i>0 && temp.charAt(i)==temp.charAt(i-1)){
            continue;
         }
         else if(temp.charAt(i)==' '){
            len++;
         }
      }
      return len;
   }
}